central/detection/lifecycle/manager_impl.go

```diff
@@ -165,7 +170,32 @@ func (m *managerImpl) flushBaselineQueue() {
 		// NOTE:  This is the only place from which Pull is called.
 		deployment := m.deploymentObservationQueue.Pull()
 
-		m.addBaseline(deployment.DeploymentID)
+		baselines := m.addBaseline(deployment.DeploymentID)
+
+		userLock := features.AutoLockProcessBaselines.Enabled()
+
+		if !userLock {
+			continue
+		}
+
+		for _, baseline := range baselines {
+			if baseline == nil {
+				continue
+			}
+			if baseline.GetUserLockedTimestamp() == nil {
+				baseline.UserLockedTimestamp = protocompat.TimestampNow()
+			}
```

Comment:
Could we do this instead:

```suggestion
			if baseline.GetUserLockedTimestamp() != nil {
				continue
			}
			baseline.UserLockedTimestamp = protocompat.TimestampNow()
```